Sail/src/Sail/entities/systems/LevelGeneratorSystem/LevelGeneratorSystem.cpp

```diff
@@ -799,136 +806,103 @@ void LevelGeneratorSystem::addDoors() {
-		//				a.sizey = 2;
-		//				possibleDoors.emplace_back(a);
-		//			}
-		//		}
-		//		if (rekt.posy - 1 >= 0 && rekt.hasDoor % 8 / 4 == 0) {
-		//			for (int i = 0; i < rekt.sizex; i++) {
-		//				rect a;
-		//				a.posx = rekt.posx + i;
-		//				a.posy = rekt.posy;
-		//				a.sizex = 1;
-		//				a.sizey = 4;
-		//				possibleDoors.emplace_back(a);
-		//			}
-		//		}
-		//		if (rekt.posy + rekt.sizey <= map->ysize && rekt.hasDoor % 2 == 0) {
-		//			for (int i = 0; i < rekt.sizex; i++) {
-		//				rect a;
-		//				a.posx = rekt.posx + i;
-		//				a.posy = rekt.posy + rekt.sizey;
-		//				a.sizex = 1;
-		//				a.sizey = 1;
-		//				possibleDoors.emplace_back(a);
-		//			}
-		//		}
-		//		if (possibleDoors.size() > 0) {
-		//			rect door = possibleDoors[rand() % possibleDoors.size()];
-		//			possibleDoors.clear();
-		//			if (door.sizey == 1) {
-		//				map->tileArr[door.posx][door.posy][2] += 1;
-		//				map->tileArr[door.posx][door.posy + 1][2] += 4;
-		//				rekt.hasDoor += 1;
-		//			}
-		//			if (door.sizey == 2) {
-		//				map->tileArr[door.posx][door.posy][2] += 2;
-		//				map->tileArr[door.posx + 1][door.posy][2] += 8;
-		//				rekt.hasDoor += 2;
-		//			}
-		//			if (door.sizey == 4) {
-		//				map->tileArr[door.posx][door.posy][2] += 4;
-		//				map->tileArr[door.posx][door.posy - 1][2] += 1;
-		//				rekt.hasDoor += 4;
-		//			}
-		//			if (door.sizey == 8) {
-		//				map->tileArr[door.posx][door.posy][2] += 8;
-		//				map->tileArr[door.posx - 1][door.posy][2] += 2;
-		//				rekt.hasDoor += 8;
-		//			}
-		//		}
-		//	}
-		//	map->matched.pop();
-		//	map->matched.push(rekt);
-		//}
-		
-		
-		//adds doors on random to add more doors
-		//only adds a door if two tiles aren't in the same "area" AND there isn't already a door there AND rand()%100 is beilow doorModifier
-		for(int i = 0; i < map->xsize; i++) {
-			for (int j = 0; j < map->ysize; j++) {
-				if (i + 1 < map->xsize) {
-					if (map->tileArr[i][j][1] != map->tileArr[i + 1][j][1] && (map->tileArr[i][j][2] % 4) / 2 == 0 && rand() % 100 < map->doorModifier) {
-						map->tileArr[i][j][2] += 2;
-						map->tileArr[i + 1][j][2] += 8;
+		// Adds a second door to each room
+		for (int c = 0; c < maxSize; c++) {
+			std::vector<rect> possibleDoors;
+			rect rekt = map->matched.front();
+			int doorCounter = 0;
+			bool up = false, right = false, down = false, left = false;
+			// Check each room which walls have doors, and how many
+			for (int i = 0; i < rekt.sizex; i++) {
+				for (int j = 0; j < rekt.sizey; j++) {
+					if (map->tileArr[rekt.posx + i][rekt.posy + j][2] % 2 == 1) {
+						up = true;
+						doorCounter++;
+					}
+					if(map->tileArr[rekt.posx + i][rekt.posy + j][2] % 4 > 1){
+						right = true;
+						doorCounter++;
+					}
+					if (map->tileArr[rekt.posx + i][rekt.posy + j][2] % 8 > 3 ) {
+						down = true;
+						doorCounter++;
+					}
+					if (map->tileArr[rekt.posx + i][rekt.posy + j][2] > 7) {
+						left = true;
+						doorCounter++;
+					}
```

Comment:
Create helper functions to make code more readable and possibly reuse the code.
I.e. bool hasLeftDoor(map, tile, ....)